### PR TITLE
[video.js] video.js types for videojs.extend

### PR DESCRIPTION
**Pull request by joeflateau**
Closes #57267

### CODE REVIEW

File: types/video.js/index.d.ts [hunk 1]
```diff
@@ -453,6 +453,19 @@ declare namespace videojs {
...
+     * Used to subclass an existing class by emulating ES subclassing using the extends keyword.
+     * @param superClass super component to extend
+     * @param subClassMethods methods sub class will add to super
+     */
+    function extend<
+        TSuper extends new (...args: any[]) => any,
+        TSubClassMethods extends Record<string | symbol, (this: InstanceType<TSuper>, ...args: any[]) => any>,
+    >(
+        superClass: TSuper,
+        subClassMethods: TSubClassMethods,
```
Comment (peterblazejewicz) [inline types/video.js/index.d.ts:466]:
```suggestion
        subClassMethods?: TSubClassMethods,
```
https://docs.videojs.com/extend.js.html#line29

**Comment by joeflateau** [comment]
good catch @peterblazejewicz

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@joeflateau thx!

File: types/video.js/index.d.ts [hunk 2]
```diff
@@ -453,6 +453,19 @@ declare namespace videojs {
      */
     function use(type: string, middleware: (player: Player) => Middleware): void;
 
+    /**
+     * Used to subclass an existing class by emulating ES subclassing using the extends keyword.
+     * @param superClass super component to extend
+     * @param subClassMethods methods sub class will add to super
```
Comment (peterblazejewicz) [inline types/video.js/index.d.ts:459]:
```suggestion
     * @param subClassMethods methods sub class will add to super
```
```suggestion
     * @param [subClassMethods] methods sub class will add to super
```

Comment (peterblazejewicz) [reply types/video.js/index.d.ts:459]: sry, editing this in place to avoid another delay

**Review by peterblazejewicz — APPROVED** [review]

**Comment by joeflateau** [comment]
Ready to merge

**Pull request merged**